Makefile

```diff
@@ -61,6 +61,7 @@ requirements: requirements/requirements.txt
 .PHONY: test
 test:
 	pytest -n auto -v --cov=mitiq --cov-report=term --cov-report=xml --ignore=mitiq/interface/mitiq_pyquil
+	pytest -rP mitiq/tests/test_about.py
```

Comment:
Why is this needed? If I run the following command to list all the tests that will be run, `test_about.py::test_stdout` is included.
```bash
pytest --collect-only -n auto -v --cov=mitiq --cov-report=term --cov-report=xml --ignore=mitiq/interface/mitiq_pyquil

Reply:
It's printing the output to verify it works. 

![image](https://github.com/unitaryfund/mitiq/assets/66048318/d881a0be-6a3d-4c62-8aaa-70e78bbf6d17)

Reply:
I see that `-rP` means that extra info will be printed when tests pass **and** have output, but

1. I don't think the test is verifying there _is_ output,[^1] and
2. I don't think we need this to be printed every time `make test` is run. As long as pytest does in fact that the printing works, we don't need to see it :)

In fact won't `test_stdout` be run twice with this PR in the current state?

[^1]: I changed the `print` statement in `about()` to `return 0` and the test still passed.